### PR TITLE
merge RootObject in with ast_node.d and astbase.d

### PR DESCRIPTION
**Pull request by WalterBright**
root/rootobject.d originally was meant as the root object for a general purpose library. It turns out that wasn't a good design, and the only use for RootObject was as the ancestor to ASTNode. The `root` directory was for general purpose modules, not dmd specific ones. So I removed it from `root`, and just pasted it in where it was used in astbase.d and ast_node.d.

The removal of the static constructor in mustuse.d was necessitated by the moving of RootObject causing a cyclic constructor dependency.

### CODE REVIEW

File: compiler/src/dmd/errors.d [hunk 1]
```diff
@@ -105,6 +105,7 @@ class ErrorSinkCompiler : ErrorSink
 /**
  * Color highlighting to classify messages
  */
+alias Color = dmd.console.Color;
```
Comment (thewilsonator) [inline compiler/src/dmd/errors.d:108]: ???

File: compiler/src/dmd/mustuse.d [hunk 2]
```diff
@@ -18,17 +18,6 @@ import dmd.globals;
 import dmd.identifier;
 import dmd.location;
 
-// Used in isIncrementOrDecrement
```
Comment (thewilsonator) [inline compiler/src/dmd/mustuse.d:21]: all the changes in this file look unrelated.

Comment (WalterBright) [reply compiler/src/dmd/mustuse.d:21]: As mentioned in the opening comment, this was to fix a cyclic static constructor dependency induced by this reorganization.

File: compiler/src/dmd/errors.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/errors.d:108]: This problem baffles me:
```
src\dmd\dtemplate.d(7910): Error: function `dmd.dtemplate.TemplateMixin.accept` does not override any function
src\dmd\dtoh.d(309):        called from here: `generateMembers()`
```
which pops up here and is the same issue I found with the Color thing above works around. Something about `generateMembers()`

Comment (WalterBright) [reply compiler/src/dmd/errors.d:108]: If I take out line 108,
```
src/dmd/errors.d(111): Error: undefined identifier `Color`
src/dmd/dtoh.d(309):        called from here: `generateMembers()`
```

Comment (WalterBright) [reply compiler/src/dmd/errors.d:108]: Hmm, I think forward references due to cyclical imports is causing this problem.

File: compiler/src/dmd/mustuse.d (hunk 2 above)
Comment (ibuclaw) [reply compiler/src/dmd/mustuse.d:21]: Because `Loc.initial` is just `Loc(0,0,0)`, maybe add a ctor for StringExp _without_ `const ref Loc`?

**Comment by WalterBright** [comment]
I'm giving up on this for now in favor of https://github.com/dlang/dmd/pull/15703
